File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.java
class Solution {
    public int findMin(int[] nums) {
        int l=0; 
        int r=nums.length-1;
        int mid=-1;
        while(l<r){
            mid=(l+r-1)/2;
            if (nums[r]<nums[mid]){
                l=mid+1;
            }
            else{
                r=mid;
            }
            
        }
        return nums[l];
        
    }
}

File: Data Structures & Algorithms/binary-search/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left = 0
        right = len(nums)-1
        while(left<=right):
            middle=(left+right)//2
            mynum = nums[middle]
            if (mynum == target):
                return middle
            elif(mynum>target):
                right = middle-1
            else:
                left = middle+1
        return -1

        